src/cdk/a11y/fake-events.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/** Returns whether an event could be a faked `mousedown` event dispatched by a screen reader. */
+export function isFakeMousedownFromScreenReader(event: MouseEvent): boolean {
+  // We can typically distinguish between these faked mousedown events and real mousedown events
+  // using the "buttons" property. While real mousedowns will indicate the mouse button that was
+  // pressed (e.g. "1" for the left mouse button), faked mousedowns will usually set the property
+  // value to 0.
+  return event.buttons === 0;
+}
+
+/** Returns whether an event could be a faked `touchstart` event dispatched by a screen reader. */
+export function isFakeTouchstartFromScreenReader(event: TouchEvent): boolean {
+  const touch: Touch | undefined = (event.touches && event.touches[0]) ||
```

Comment:
Could you add a comment here capturing the screen-reader + browser combinations which exhibit the detected behavior here (including versions)?